app/javascript/controllers/confirmation_controller.js

```diff
@@ -0,0 +1,108 @@
+import { Controller } from "@hotwired/stimulus"
+
+/**
+ * Connects to data-controller="confirmation"
+ * Displays a confirmation modal with the details of the form that user just submitted.
+ * Launched when the user clicks Save from the form.
+
+ * First runs a "pre-check" on the form data to a validation endpoint,
+ * which is specified in the controller's `preCheckPathValue` property.
+ * If the pre-check passes, it shows the modal. Because the confirmation modal should only be shown
+ * when the form data can pass initial validation.
+ * If the pre-check fails, it submits the form to the server for full validation and render with the errors.
+ *
+ * The pre-check validation endpoint also returns the html body to display in the modal if validation passes.
+
+ * If the user clicks the "Yes..." button from the modal, it submits the form.
+ * If the user clicks the "No..." button from the modal, it closes and user remains on the same url.
+ */
+export default class extends Controller {
+  static targets = [
+    "modal",
+    "form"
+  ]
+
+  static values = {
+    preCheckPath: String
+  }
+
+  openModal(event) {
+    event.preventDefault();
+
+    const formData = new FormData(this.formTarget);
+    const formObject = this.buildNestedObject(formData);
+
+    fetch(this.preCheckPathValue, {
+      method: "POST",
+      headers: {
+        "X-CSRF-Token": this.getMetaToken(),
+        "X-Requested-With": "XMLHttpRequest",
+        "Content-Type": "application/json",
+        Accept: "application/json"
```

Comment:
Nitpick - can we put quotes around "Accept" to make it match the rest of the object?

Reply:
Sure, I can also rebase while I'm at it, in case @cielf will be retesting for functionality.

Reply:
Quotes added.